tasks/validate_rbac_token.rb

```diff
@@ -0,0 +1,77 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require 'uri'
+require 'net/https'
+require 'json'
+require 'etc'
+require 'puppet'
+
+# Class to check an rbac token is valid
+class ValidateRbacToken
+  def initialize(params)
+    @token_file = params['token_file']
+  end
+
+  def execute!
+    token_file = @token_file || File.join(Etc.getpwuid.dir, '.puppetlabs', 'token')
+
+    uri = URI("https://#{Puppet.settings[:certname]}:4433/rbac-api/v2/auth/token/authenticate")
+    https = https_object(uri: uri)
+    request = request_object(token_file: token_file)
+
+    resp = https.request(request)
+
+    if resp.code == '200'
+      puts 'RBAC token is valid'
+      exit 0
+    else
+      body = JSON.parse(resp.body)
+      case resp.code
+      when '401', '403'
+        puts "#{resp.code} #{body['kind']}: \nCheck your API token at #{token_file}. \
+              An alternate token file can be specified using the token_file param. \n\n\
+              See https://www.puppet.com/docs/pe/latest/rbac_token_auth_intro for more details. \n"
```

Comment:
```suggestion
        puts "#{resp.code} #{body['kind']}: " \
             "Check your API token at #{token_file}.\n" \
             "Please refresh your token or provide an alternate file.\n" \
             "See https://www.puppet.com/docs/pe/latest/rbac_token_auth_intro for more details.\n"
```
Changing formatting a bit, and asking to refresh the token